Model/ModelManager.php

```diff
@@ -385,8 +386,12 @@ public function getNormalizedIdentifier($entity)
             throw new \RunTimeException('Invalid argument, object or null required');
         }
 
-        // the entities is not managed
-        if (!$entity || !$this->getEntityManager($entity)->getUnitOfWork()->isInIdentityMap($entity)) {
+        if (!$entity) {
+            return;
+        }
+
+        $entityState = $this->getEntityManager($entity)->getUnitOfWork()->getEntityState($entity);
+        if (UnitOfWork::STATE_NEW == $entityState || UnitOfWork::STATE_REMOVED == $entityState) {
```

Comment:
I second the strict comparison argument, as it can be seen as unnecessary, distracting noise. As for the performance argument, I think we should optimize for readability, and this piece of code is not called intensively, AFAIK